TicketsDemo.Domain/DefaultImplementations/Logger.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.IO;
+using TicketsDemo.Domain.Interfaces;
+
+namespace TicketsDemo.Domain.DefaultImplementations
+{
+    public class Logger : ILogger
+    {
+        private string _fileName;
+        public Logger(string fileName)
+        {
+            _fileName = fileName;
+        }
+        public void Log(string message, LogSeverity severity)
+        {
+            string fullPath = _fileName + "\\DebugLog.txt";
+
+            try
+            {
+                using (StreamWriter sw = new StreamWriter(fullPath, true, System.Text.Encoding.Default))
+                {
+                    var wrtStr = String.Format("{0}[{1}]: {2}", severity, DateTime.Now, message);
+                    sw.WriteLine(wrtStr);
+                }
+            }
+            catch (Exception ex)
+            {
+                fullPath = _fileName + "\\DebugError.txt";
+                using (StreamWriter sw = new StreamWriter(fullPath, true, System.Text.Encoding.Default))
+                {
+                    var wrtStr = String.Format("\n{1}\nError: {0} \n ", ex, DateTime.Now);
+                    sw.WriteLine(wrtStr);
+                }
+            }
```

Comment:
when exception is caught inside the logger - it 'have to' be re-thrown if logged message is of high severity (such as Warn or Error or Fatal).

This enforces system-wide event monitoring mechanics, so system administrator has a chance to track this issue